receiver/sqlserverreceiver/scraper.go

```diff
@@ -470,3 +490,258 @@ func (s *sqlServerScraperHelper) recordDatabaseStatusMetrics(ctx context.Context
+			s.logger.Error(fmt.Sprintf("sqlServerScraperHelper failed parsing open transaction count. original value: %s, err: %s", row[openTransactionCount], err))
+		}
+		transactionIDVal, err := strconv.Atoi(row[transactionID])
+		if err != nil {
+			s.logger.Error(fmt.Sprintf("sqlServerScraperHelper failed parsing transaction id number. original value: %s, err: %s", row[transactionID], err))
+		}
+		// percent complete and estimated completion time is a real value in mssql
+		percentCompleteVal, err := strconv.ParseFloat(row[percentComplete], 32)
+		if err != nil {
+			s.logger.Error(fmt.Sprintf("sqlServerScraperHelper failed parsing percent complete. original value: %s, err: %s", row[percentComplete], err))
+		}
+		estimatedCompletionTimeVal, err := strconv.ParseFloat(row[estimatedCompletionTime], 32)
+		if err != nil {
+			s.logger.Error(fmt.Sprintf("sqlServerScraperHelper failed parsing estimated completion time number. original value: %s, err: %s", row[estimatedCompletionTime], err))
+		}
+		cpuTimeVal, err := strconv.Atoi(row[cpuTime])
+		if err != nil {
+			s.logger.Error(fmt.Sprintf("sqlServerScraperHelper failed parsing cpu time number. original value: %s, err: %s", row[cpuTime], err))
+		}
+		totalElapsedTimeVal, err := strconv.Atoi(row[totalElapsedTime])
+		if err != nil {
+			s.logger.Error(fmt.Sprintf("sqlServerScraperHelper failed parsing total elapsed time. original value: %s, err: %s", row[totalElapsedTime], err))
+		}
+		readsVal, err := strconv.Atoi(row[reads])
+		if err != nil {
+			s.logger.Error(fmt.Sprintf("sqlServerScraperHelper failed parsing read count. original value: %s, err: %s", row[reads], err))
+		}
+		writesVal, err := strconv.Atoi(row[writes])
+		if err != nil {
+			s.logger.Error(fmt.Sprintf("sqlServerScraperHelper failed parsing write count. original value: %s, err: %s", row[writes], err))
+		}
+		logicalReadsVal, err := strconv.Atoi(row[logicalReads])
+		if err != nil {
+			s.logger.Error(fmt.Sprintf("sqlServerScraperHelper failed parsing logical read count. original value: %s, err: %s", row[logicalReads], err))
+		}
+		transactionIsolationLevelVal, err := strconv.Atoi(row[transactionIsolationLevel])
+		if err != nil {
+			s.logger.Error(fmt.Sprintf("sqlServerScraperHelper failed parsing transaction isolation level. original value: %s, err: %s", row[transactionIsolationLevel], err))
+		}
+		// todo: check case sensitive?
+		lockTimeoutVal := 0
+		if row[lockTimeout] != "" {
+			lockTimeoutVal, err = strconv.Atoi(row[lockTimeout])
+			if err != nil {
+				s.logger.Error(fmt.Sprintf("sqlServerScraperHelper failed parsing lock timeout. original value: %s, err: %s", row[lockTimeout], err))
+			}
+		}
+
+		deadlockPriorityVal := 0
+		if row[deadlockPriority] != "" {
+			deadlockPriorityVal, err = strconv.Atoi(row[deadlockPriority])
+			if err != nil {
+				s.logger.Error(fmt.Sprintf("sqlServerScraperHelper failed parsing deadlock priority. original value: %s, err: %s", row[deadlockPriority], err))
+			}
+		}
```

Comment:
Could I know why the coding style is different than other line to parsing int?

Other lines are like

```go
transactionIsolationLevelVal, err := strconv.Atoi(row[transactionIsolationLevel])
if err != nil {
	s.logger.Error(fmt.Sprintf("sqlServerScraperHelper failed parsing transaction isolation level. original value: %s, err: %s", row[transactionIsolationLevel], err))
}
```

Reply:
updated